wear/src/main/java/com/example/util/simpletimetracker/presentation/navigation/MainNavigator.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package com.example.util.simpletimetracker.presentation.navigation
+
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.wear.compose.navigation.SwipeDismissableNavHost
+import androidx.wear.compose.navigation.composable
+import androidx.wear.compose.navigation.rememberSwipeDismissableNavController
+import com.example.util.simpletimetracker.presentation.screens.ActivitiesScreen
+import com.example.util.simpletimetracker.presentation.screens.TagsScreen
+
+@Composable
+fun MainNavigator() {
+    val navController = rememberSwipeDismissableNavController()
+    SwipeDismissableNavHost(
+        navController = navController,
+        startDestination = "activities"
+    ) {
+        composable("activities") {
+            ActivitiesScreen(onSelectActivity = { id ->
+                navController.navigate("activities/$id/tags")
+            })
+        }
+        composable("activities/{id}/tags") {
+            TagsScreen(activityId = it.arguments?.getString("id")?.toLong()!!, onSelectTag = {
+                navController.navigate("activities")
+            })
```

Comment:
Looking at this again... how does this structure lend itself to setting the current activities at tag select time, considering that the setCurrentActivities API needs to know the full list of current activities? It would seem that the TagsScreen would need to introduce its own currentActivities model object so that it can add the selected activity to the list of current ones, but that would introduce its own set of RPC network calls to get that model object until the RPC client is made stateful with a cache.

Reply:
This extra set of network calls isn't a problem knowing that the plan is to add that caching eventually. More want to check that my impression of how to develop in this presentation layer aligns with yours.

Reply:
Good points!

For this first pass, my focus was solely on foundational navigation so I didn't spend too much time here.

From other projects, I know navigation libraries usually either have support for custom payloads OR have a recommended approach for passing more than primitives from one route to another.

According to Google's docs, it looks like the recommendation is exactly what you identified: [add more performance optimizations in the Data Layer.](https://developer.android.com/jetpack/compose/navigation#retrieving-complex-data)